src/stores/search.ts

```diff
@@ -0,0 +1,341 @@
+    providers,
+  }: {
+    mediaType: SupportedMediaType
+    providers: { source_name: string; display_name: string }[]
+  }) {
+    const providersKey: FilterCategory = `${mediaType}Providers`
+    const currentProviders = state.filters[providersKey]
+      ? [...state.filters[providersKey]]
+      : []
+    state.filters[providersKey] = providers.map((provider) => {
+      const existingProviderFilterIdx = currentProviders.findIndex(
+        (p) => p.code === provider.source_name
+      )
+      const checked =
+        existingProviderFilterIdx >= 0
+          ? currentProviders[existingProviderFilterIdx].checked
+          : false
+
+      return {
+        code: provider.source_name,
+        name: provider.display_name,
+        checked,
+      }
+    })
+  }
+  /**
+   * Toggles a filter's checked parameter. Requires either codeIdx or code.
+   */
+  function toggleFilter(params: {
+    filterType: import('../store/types').FilterCategory
+    codeIdx?: number
+    code?: string
+  }) {
+    if (
+      typeof params.codeIdx === 'undefined' &&
+      typeof params.code === 'undefined'
+    ) {
+      warn(
+        `Cannot toggle filter of type ${params.filterType}. Use code or codeIdx parameter`
+      )
+      return
+    }
+    const { filterType } = params
+    const codeIdx =
+      params.codeIdx ??
+      state.filters[filterType].findIndex((f) => f.code === params.code)
+    state.filters[filterType][codeIdx].checked =
+      !state.filters[filterType][codeIdx].checked
+  }
+
+  /**
+   * Resets all filters to initial values.
+   * Provider filters are not in the initial filters, so they need to be
+   * handled separately.
+   *
+   */
+  const clearFilters = () => replaceFilters(getBaseFiltersWithProviders())
+
+  /**
+   * Selecting some filter items disables related items. For example, selecting an `nc`
+   * license filter (CC BY-NC, CC BY-NC-SA, CC BY-NC-ND) disables the `Commercial` license
+   * type filter. This function determines if the filter item should be disabled based on
+   * the currently checked filter items.
+   */
+  function isFilterDisabled(
+    item: FilterItem,
+    filterCategory: FilterCategory
+  ): boolean | undefined {
+    if (!['licenseTypes', 'licenses'].includes(filterCategory)) {
+      return
+    }
+    if (item.code === 'commercial' || item.code === 'modification') {
+      const targetCode = {
+        commercial: 'nc',
+        modification: 'nd',
+      }[item.code]
+      return state.filters.licenses.some(
+        (item) => item.code.includes(targetCode) && item.checked
+      )
+    } else {
+      const dependentFilters: string[] = []
+      if (item.code.includes('nc')) {
+        dependentFilters.push('commercial')
+      }
+      if (item.code.includes('nd')) {
+        dependentFilters.push('modification')
+      }
+      return (
+        state.filters.licenseTypes.filter(
+          (item) => dependentFilters.includes(item.code) && item.checked
+        ).length > 0
```

Comment:
nit: filtering for existence could be `some` and save iterating the full list (if this is simpler, YMMV).

```suggestion
        state.filters.licenseTypes.some(
          (item) => dependentFilters.includes(item.code) && item.checked
        )
```